cordova-common/src/CordovaLogger.js

```diff
@@ -153,6 +153,30 @@ CordovaLogger.prototype.setLevel = function (logLevel) {
 };
 
 /**
+ * Adjusts the current logger level according to the passed options.
+ *
+ * @param   {Object|Array}  opts  An object or args array with options
+ *
+ * @return  {CordovaLogger}     Current instance, to allow calls chaining.
+ */
+CordovaLogger.prototype.adjustLoggerLevel = function (opts) {
+    if (opts instanceof Array) {
+        opts.silent = opts.indexOf('--silent') !== -1;

```

Comment:
Please do not change the type of the object. Why not go with something concise and easy to understand like : 
```
if(Array.isArray(opts)) {
     if(opts.indexOf('--verbose') > -1) {
        this.setLevel('verbose');
    }
    else if (opts.indexOf('--silent') > -1) {
        this.setLevel('error');
    }
}
return this;
```

Note also that I reversed the order of checks in the if/else to reflect that in the original code --verbose out ranks --silent